BlueprintUI/Sources/Layout/Layout.swift

```diff
@@ -33,10 +129,172 @@ public protocol Layout {
 
 }
 
-extension Layout where Traits == () {
+extension LegacyLayout where Traits == () {
 
     public static var defaultTraits: () {
         return ()
     }
 
 }
+
+public protocol CaffeinatedLayout {
+
+    typealias Subelements = LayoutSubelements
+
+    /// Cached values associated with the layout instance.
+    ///
+    /// If you create a cache for your custom layout, you can use a type alias to define this type
+    /// as your data storage type. Alternatively, you can refer to the data storage type directly in
+    /// all the places where you work with the cache.
+    ///
+    /// See ``makeCache(subelements:)-4kmy2`` for more information.
+    associatedtype Cache = Void
+
+    /// Returns the size of the composite element, given a proposed size constraint and the
+    /// container's subelements.
+    ///
+    /// Implement this method to tell your custom layout container’s parent how much space the
+    /// container needs for a set of subelements, given a size constraint. The parent might call
+    /// this method more than once during a layout pass with different proposed sizes to test the
+    /// flexibility of the container.
+    ///
+    /// In Blueprint, parents ultimately choose the size of their children, so the actual size that
+    /// this container is laid out in may not be a size that was returned from this method.
+    ///
+    /// ## Sizing Rules
+    ///
+    /// For performance reasons, the layout engine may deduce the measurement of your container for
+    /// some constraint values without explicitly calling
+    /// ``sizeThatFits(proposal:subelements:cache:)``. To ensure that the deduced value is correct,
+    /// your layout must follow some ground rules:
+    ///
+    /// 1. **Given one fixed constraint axis, the element's growth along the other axis should be
+    ///    _monotonic_.** That is, an element can grow when given a larger constraint, or shrink
+    ///    when given a smaller constraint, but it should never shrink when given a larger
+    ///    constraint. When growing on one axis, it is OK to shrink along the other axis, such as a
+    ///    block of text that re-flows as the width changes.
+    /// 2. If your element has no intrinsic size along an axis, you can represent that in a couple
+    ///     ways:
+    ///     - You can return a fixed value representing the minimum size for your element. In this
+    ///       approach, a containing element is usually responsible for stretching your element to
+    ///       fill desired space.
+    ///     - You can return a size that entirely fills the constraint proposed. In this approach,
+    ///       you **must** return `.infinity` when the constraint is
+    ///       ``SizeConstraint/Axis/unconstrained``. Otherwise, your behavior would be in violation
+    ///       of rule #1.
+    ///
```

Comment:
This is excellent. It may be accompanied by links to further documentation which show examples of the convention in practice as well as definitions which break the convention and the problems that such cause.

Reply:
Sure, good idea. I'm working on some supplemental documentation with diagrams.